### ISSUE TITLE
PostgreSQL: parsing error attempting to revoke role from another role

### ISSUE THREAD

**Issue by knpwrs** [sqlfluff/sqlfluff#3907]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

sqlfluff says `1 templating/parsing error found` with the following line:

```sql
REVOKE lc_anonymous FROM lc_api;
```

This is valid in PostgreSQL.

### Expected Behaviour

Parsing should succeed as this is a valid statement (see https://www.postgresql.org/docs/current/sql-revoke.html#SQL-REVOKE-EXAMPLES).

### Observed Behaviour

Parsing fails.

### How to reproduce

Attempt to run sqlfluff against file containing offending line.

### Dialect

Postgres

### Version

I am using the `sqlfluff/sqlfluff:1.3.2` (the official docker image and whatever verison of python it is using).

### Configuration

```ini
[sqlfluff]
dialect = postgres

[sqlfluff:rules]
tab_space_size = 2
indent_unit = space

[sqlfluff:rules:L057]
ignore_words = "uuid-ossp"
```

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
